packages/@react-spectrum/list/src/ListViewItem.tsx

```diff
@@ -163,110 +162,112 @@ export function ListViewItem<T>(props: ListViewItemProps<T>) {
   }
 
   return (
-    <div
-      {...mergedProps}
-      className={
-        classNames(
-          listStyles,
-          'react-spectrum-ListView-row',
-          {
-            'focus-ring': isFocusVisible,
-            'round-tops':
-              roundTops || (isHovered && !isSelected && state.selectionManager.focusedKey !== item.key),
-            'round-bottoms':
-              roundBottoms || (isHovered && !isSelected && state.selectionManager.focusedKey !== item.key)
-          }
-        )
+    <>
+      {isListDroppable &&
+        <div role="button" {...visuallyHiddenProps} {...dropIndicator?.dropIndicatorProps} ref={dropIndicatorRef} />
```

Comment:
Thought process:
- We want to streamline the mobile screen reader DnD experience in ListView. Currently, if a row is a valid drop target, the content inside the row (e.g. checkbox, text, draghandle) is accessible by the screenreader since the row is not aria-hidden

Solutions:
- hide all content within a droppable row. Roughly implemented in a prior commit [059d91b](https://github.com/adobe/react-spectrum/pull/3401/commits/059d91b5da4bf84bf58eac0c9fca318b4847516a). Felt brittle
- move the drop indicator button outside the row and always hide the row itself during a drag session except if it contains the original drag handle (current implementation). May not work super well in the iOS VO case since aria-hidden doesn't work when applied to role="row" (perhaps just need to apply aria-hidden to the gridcell too?)

Reply:
TODO: what about the iOS VO case since aria-hidden on the row doesn't actually hide the row from VO? Maybe move all drop indicators (root, insertion, and on) outside the Virtualizer's presentation div so ariaHideOutside can apply hidden to the presentation div?

Reply:
Pretty sure this will result in an invalid accessibility tree though. `role=grid` only accepts elements with `role=row -> role=gridcell` as children, it can't have buttons floating inside there without a row/cell parent.

Reply:
But shouldn't we already hide everything other than the drop indicator, including siblings of it? I see you got rid of `useDroppableItem`, which was previously registering the `rowRef` as a drop target as well, which would have prevented it from being hidden.

Reply:
> Pretty sure this will result in an invalid accessibility tree though. `role=grid` only accepts elements with `role=row -> role=gridcell` as children, it can't have buttons floating inside there without a row/cell parent.

Urgh, what if I wrapped the drop indicator button inside a row + gridcell? Dunno if that would them make the screenreader user have to swipe past that row and gridcell as well then, something I'll try out.


> But shouldn't we already hide everything other than the drop indicator, including siblings of it? I see you got rid of `useDroppableItem`, which was previously registering the `rowRef` as a drop target as well, which would have prevented it from being hidden.

The below is in context of the "drag into folder" story:
So there were a couple of issues beyond the extraneous useDroppableItem. Since the drop indicator button isn't actually rendered until the keyboard/screenreader drag session starts, the drop targets don't get registered by DragManager and thus none of the rows are aria  when the drag session starts. 

When I got rid of that conditional render the drop targets are all registered properly and the rows are properly hidden for the most part. However, the drop indicator becomes one of the rows elements that the user can get to via the left/right arrow since it is always rendered (can be fixed via changing useGridCell's focusableTreeWalker options perhaps). Additionally, the user still has to swipe twice to go to the next droppable target: one swipe brings you onto the row, the second onto the drop indicator button.

IMO ideally, I would make the row the drop indicator and have the grid cell always be aria-hidden during a keyboard/screenreader operation so that the user only swipes through the rows themselves, with a sole exception being the gridcell containing the original dragged key. It would take some tinkering since useDropIndicator applies aria-hidden="true" if we aren't in a drag session (would need to communicate that somehow).

Moving the drop indicator outside of the row felt easier to me since I wouldn't need to do too much extra logic, but the accessibility tree bit you brought up is a good point

Reply:
> Since the drop indicator button isn't actually rendered until the keyboard/screenreader drag session starts, the drop targets don't get registered by DragManager and thus none of the rows are aria when the drag session starts.

That sounds like the root of the issue. `registerDropItem` in DragManager should probably call `updateValidDropTargets` like `registerDropTarget` does so that the drop target list is up to date.

Reply:
Looking at it again, I think I can just get rid of the `isDropTarget` from https://github.com/adobe/react-spectrum/blob/main/packages/%40react-spectrum/list/src/ListViewItem.tsx#L229 and that should make the drop indicator for valid drop targets render and register properly.

However, this will still mean that the row itself can be "focused" by the mobile screenreaders along with the drop indicators resulting in the double swipe required to navigate between valid droppable rows (one of the main reasons why I was trying to move the drop indicator out of the row. For clarification, are we fine with that? If not I can see about making the row itself the dropIndicator like I mentioned above

Reply:
Ah I see. I think that's only on TalkBack. iOS VoiceOver will never focus the row itself.